### PR TITLE
[strategy] Update colossalai strategy

### PR DESCRIPTION
**Pull request by 1SAA**
## What does this PR do?
Update the APIs in ColossalAIStrategy for its new version.
Since the API of Gemini in ColossalAI has updated recently, it becames incompatible with Lightning.
For compatibility with the upcoming version 0.1.11, I updated the ColossalAIStrategy.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/strategies/colossalai.py [hunk 1]
```diff
@@ -264,23 +266,52 @@ def setup_precision_plugin(self) -> None:
...
-                GeminiManager.get_default_device(self.placement_policy),
-            )
-            gemini_manager = GeminiManager(self.placement_policy, chunk_manager)
-            model = _LightningModuleWrapperBase(self.model)
-            self.model = ZeroDDP(model, gemini_manager, self.force_outputs_fp32)
+                with _patch_cuda_is_available():
+                    from colossalai.nn.parallel import GeminiDDP
+                    from colossalai.utils import get_current_device
+
+                assert self.use_chunk, "`ColossalAIStrategy` must use chunk in versions higher than 0.1.10"
```
Comment (tchaton) [inline src/pytorch_lightning/strategies/colossalai.py:295]: Let's use proper error handling ;)

**Review by tchaton — COMMENTED** [review]
Mind adding some unit tests ?

File: src/pytorch_lightning/strategies/colossalai.py [hunk 2]
```diff
@@ -264,23 +266,52 @@ def setup_precision_plugin(self) -> None:
                 )
         assert isinstance(self.model, (pl.LightningModule, _LightningPrecisionModuleWrapperBase))
         pl_module = self.model
-        process_group = ProcessGroup()
+
+        new_version_flag = compare_version("colossalai", operator.ge, "0.1.10")
+
```
Comment (awaelchli) [inline src/pytorch_lightning/strategies/colossalai.py:271]: How important is it to support the old code path? Shouldn't we set the min version to 0.1.10?

File: src/pytorch_lightning/strategies/colossalai.py (hunk 1 above)
Comment (1SAA) [reply src/pytorch_lightning/strategies/colossalai.py]: ok.

File: src/pytorch_lightning/strategies/colossalai.py (hunk 2 above)
Comment (1SAA) [reply src/pytorch_lightning/strategies/colossalai.py]: we should definitely support the old code path. the new version is updated for training efficiency, but it is not a quite stable version. we will fix some potential bugs and add more practical functionalities soon. since 0.1.10 is a stable version, we should keep it currently.

**Comment by 1SAA** [comment]
@tchaton It is a good suggestion. I think I can add some new unit tests.

File: src/pytorch_lightning/strategies/colossalai.py [hunk 3]
```diff
@@ -264,23 +267,54 @@ def setup_precision_plugin(self) -> None:
                 )
         assert isinstance(self.model, (pl.LightningModule, _LightningPrecisionModuleWrapperBase))
         pl_module = self.model
-        process_group = ProcessGroup()
+
+        new_version_flag = compare_version("colossalai", operator.gt, "0.1.10")
+
```
Comment (awaelchli) [inline src/pytorch_lightning/strategies/colossalai.py:272]: Could you define it at the top of the file as

```py
_COLOSSALAI_GREATER_EQUAL_0_1_10 = compare_version("colossalai", operator.gt, "0.1.10")
```

and reuse it here in the code?
We follow this convention across the codebase and it would be nice to keep it standardized.

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: src/pytorch_lightning/strategies/colossalai.py [hunk 4]
```diff
@@ -264,23 +266,50 @@ def setup_precision_plugin(self) -> None:
...
-                chunk_size,
-                process_group,
-                self.enable_distributed_storage,
-                GeminiManager.get_default_device(self.placement_policy),
-            )
-            gemini_manager = GeminiManager(self.placement_policy, chunk_manager)
-            model = _LightningModuleWrapperBase(self.model)
-            self.model = ZeroDDP(model, gemini_manager, self.force_outputs_fp32)
+                with _patch_cuda_is_available():
+                    from colossalai.nn.parallel import GeminiDDP
```
Comment (carmocca) [inline src/pytorch_lightning/strategies/colossalai.py:290]: This import failed even though GPU ci installs [`0.1.10+torch1.12cu11.3`](https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=121637&view=logs&j=3f274fac-2e11-54ca-487e-194c91f3ae9f&t=46417416-7aac-584b-0305-5d8c82c23854&l=47)

Looks like it was only added in 0.11, not 0.10 as the PR implements

Comment (1SAA) [reply src/pytorch_lightning/strategies/colossalai.py:290]: yeah. `GeminiDDP` is newly added in 0.1.12.

Comment (awaelchli) [reply src/pytorch_lightning/strategies/colossalai.py:290]: @1SAA then let's update the version check to `_COLOSSALAI_GREATER_0_1_12`?

File: src/pytorch_lightning/CHANGELOG.md [hunk 5]
```diff
@@ -30,6 +30,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added a warning when `self.log(..., logger=True)` is called without a configured logger ([#15814](https://github.com/Lightning-AI/lightning/pull/15814))
 
 
+- Added support for colossalai 0.1.10 ([#15888](https://github.com/Lightning-AI/lightning/pull/15888))
```
Comment (carmocca) [inline src/pytorch_lightning/CHANGELOG.md:33]:
```suggestion
- Added support for colossalai 0.1.11 ([#15888](https://github.com/Lightning-AI/lightning/pull/15888))
```

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
The mypy checks are failing: https://github.com/Lightning-AI/lightning/actions/runs/3711224217/jobs/6291426057
